entity_macros/src/derive/ent/struct/data/internal.rs

```diff
@@ -28,30 +32,95 @@ pub struct Ent {
 pub struct EntField {
     pub ident: Option<Ident>,
     pub ty: Type,
-    #[darling(default, rename = "id")]
-    pub is_ent_id_field: bool,
-    #[darling(default, rename = "database")]
-    pub is_ent_database_field: bool,
-    #[darling(default, rename = "created")]
-    pub is_ent_created_field: bool,
-    #[darling(default, rename = "last_updated")]
-    pub is_ent_last_updated_field: bool,
+    /// Location of the word `id`, if present.
+    #[darling(default)]
+    id: Option<SpannedValue<()>>,
+    /// Location of the word `database`, if present.
+    #[darling(default)]
+    database: Option<SpannedValue<()>>,
+    /// Location of the word `created`, if present.
+    #[darling(default)]
+    created: Option<SpannedValue<()>>,
+    /// Location of the word `last_updated`, if present.
+    #[darling(default)]
+    last_updated: Option<SpannedValue<()>>,
     #[darling(default, rename = "field")]
     pub field_attr: Option<Override<FieldAttr>>,
     #[darling(default, rename = "edge")]
     pub edge_attr: Option<EdgeAttr>,
 }
 
+impl EntField {
+    pub fn is_id_field(&self) -> bool {
+        self.id.is_some()
+    }
+
+    pub fn is_database_field(&self) -> bool {
+        self.database.is_some()
+    }
+
+    pub fn is_created_field(&self) -> bool {
+        self.created.is_some()
+    }
+
+    pub fn is_last_updated_field(&self) -> bool {
+        self.last_updated.is_some()
+    }
+
+    /// Check whether this field is trying to declare itself as multiple ent fields.
+    /// This is nonsensical, and indicates an error on the part of the caller.
+    ///
+    /// # Example
+    /// This would return true for `#[ent(id, database)]`.
+    fn declares_multiple_known_fields(&self) -> bool {
+        let mut known_fields_declared = 0;
+        if self.is_id_field() {
+            known_fields_declared += 1;
+        }
+
+        if self.is_database_field() {
+            known_fields_declared += 1;
+        }
+
+        if self.is_created_field() {
+            known_fields_declared += 1;
+        }
+
+        if self.is_last_updated_field() {
+            known_fields_declared += 1;
+        }
+
+        known_fields_declared > 1
+    }
+
+    /// Validate that at most one of `id`, `database`, `created`, and `last_updated` are declared.
+    /// If that is not the case, create properly-spanned errors for each declaration.
+    pub fn validate_zero_or_one_known_fields(&self) -> darling::Result<()> {
+        if self.declares_multiple_known_fields() {
+            let mut errors = vec![];
+            for item in vec![&self.id, &self.database, &self.created, &self.last_updated] {
```

Comment:
Looks like there was a clippy warning about converting to `&[&self.id, &self.database, &self.created, &self.last_updated]` since a slice implements IntoIterator in this case just the same.